yt_dlp/extractor/amcnetworks.py

```diff
@@ -90,27 +114,47 @@ def _real_extract(self, url):
         formats, subtitles = self._extract_theplatform_smil(
             media_url, video_id)
         self._sort_formats(formats)
+
+        thumbnails = []
+        thumbnail_temp = []
+        thumbnail_temp.append(info.get('thumbnail'))
+        thumbnail_temp.append(properties.get('imageDesktop'))
+        for thumbnail_url in thumbnail_temp:
+            if not thumbnail_url:
+                continue
+            i = {'url': thumbnail_url}
+            mobj = re.search(r'(\d+)x(\d+)', thumbnail_url)
+            if mobj:
+                i.update({
+                    'width': int(mobj.group(1)),
+                    'height': int(mobj.group(2)),
+                })
+            thumbnails.append(i)
+        # Sometimes these thumbnails are the same image despite having
+        # different URLs.  Sometimes they're entirely different images,
+        # not just different resolutions of one image.
```

Comment:
```suggestion
        thumbnails = []
        thumbnail_urls = [info.pop('thumbnail')]
        thumbnail_urls.append(properties.get('imageDesktop'))
        for thumbnail_url in thumbnail_urls:
            if not thumbnail_url:
                continue
            mobj = re.search(r'(\d+)x(\d+)', thumbnail_url)
            width, height = mobj.groups() if mobj else (None, None)
            thumbnails.append({
                'url': thumbnail_url
                'width': int_or_none(width),
                'height': int_or_none(height),
            })
```
Again, untested, just a suggestion for refactoring
1. simplified code
2. `info` shouldn't have both `thumbnail` and `thumbnails`

Reply:
> 2. `info` shouldn't have both `thumbnail` and `thumbnails`

Aha. That's an example of why I wish extractors would follow a template. Some use 'thumbnail' and some use 'thumbnails'. Some use 'info' and some return data without assigning them first to 'info'.

Reply:
I copied the lines pertaining to 'mobj' from other extractors. Should the code be kept consistent across extractors? Or should it just be defined once and imported into each extractor?

Reply:
Should the thumbnails be listed in order of increasing resolution? If so, the imageDesktop thumbnail is usually the largest one and should be appended last.

Reply:
They will get automatically get sorted by resolution later. So order doesnt matter

Reply:
What happens if the resolution can't be deduced from the file name?

Reply:
If you know that `imageDesktop` will always be higher quality, you can set it's `preference=1`